source/atomic_ops.cpp

```diff
@@ -63,7 +62,56 @@ bool atomic_cas<uint32_t>(uint32_t *ptr, uint32_t *expectedCurrentValue, uint32_
 
     return !__STREXW(desiredValue, ptr);
 }
-#endif /* #if (__CORTEX_M >= 0x03) */
+
+uint32_t atomic_incr(uint32_t *valuePtr, uint32_t delta)
+{
+    uint32_t newValue;
+    do {
+        newValue = __LDREXW(valuePtr) + delta;
```

Comment:
You're right. This should be, at minimum, handled in different PR's, since it's dealing with very different questions. I will open another PR to deal with the question of ```atomic_incr/decr``` so that others have the opportunity to comment. However, unless there is another opinion which will sway the vote, they will not be merged.